### PR TITLE
Fix meta description clash

### LINKED ISSUE #8713: Duplicate meta descriptions hurting SEO score
## Details
https://github.com/gravitational/teleport/blob/master/docs/pages/setup/reference/authentication.mdx and
https://github.com/gravitational/teleport/blob/master/docs/pages/setup/reference/license.mdx
have the same meta description; looks like a potential copy/paste error.  Please make descriptions unique to avoid SEO hit. 

### Category

- Improve Existing

### PR DESCRIPTION
**Pull request by ptgott**
Two docs pages have the same meta description, which hurts SEO.
This changes the meta description of one of the page and uses
this opportunity to do some light copy editing.

Fixes #8713

### CODE REVIEW

File: docs/pages/setup/reference/authentication.mdx [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 title: Authentication options
-description: Teleport enterprise license file configuration parameters and requirements
+description: A reference for Teleport's authentication connectors
 ---
 
-Teleport uses the concept of "authentication connectors" to authenticate users
-when they execute [`tsh login`](./cli.mdx#tsh-login) command. There are three
-types of authentication connectors:
+Teleport uses the concept of **authentication connectors** to authenticate users
+when they execute the [`tsh login`](./cli.mdx#tsh-login) command. There are three
```
Comment (zmb3) [inline docs/pages/setup/reference/authentication.mdx:7]: auth connectors are used for web UI logins as well, not just `tsh login`.
```suggestion
with an identity provider. There are three
```

File: docs/pages/setup/reference/authentication.mdx [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
+types of authentication connectors.
 
 ## Local
```
Comment (zmb3) [inline docs/pages/setup/reference/authentication.mdx:10]: If we want to be accurate, I don't think local users use an auth connector. An auth connector is for when you want to authenticate against an _external_ identity provider. If you want to authenticated against teleport's internal user DB, then you don't need to create an auth connector at all.

File: docs/pages/setup/reference/authentication.mdx [hunk 3]
```diff
@@ -1,22 +1,21 @@
...
 Local authentication is used to authenticate against a local Teleport user
-database. This database is managed by [`tctl users`](./cli.mdx#tctl-users-add)
+database. This database is managed by the [`tctl users`](./cli.mdx#tctl-users-add)
 command. Teleport also supports second-factor authentication (2FA) for the local
 connector. There are several possible values (types) of 2FA:
 
-- `otp` is the default. It implements [TOTP](https://en.wikipedia.org/wiki/Time-based_One-time_Password_Algorithm)
-  standard. You can use [Google Authenticator](https://en.wikipedia.org/wiki/Google_Authenticator)
-  or [Authy](https://www.authy.com/) or any other TOTP client.
+- `otp` is the default. It implements the [TOTP](https://en.wikipedia.org/wiki/Time-based_One-time_Password_Algorithm)
```
Comment (zmb3) [inline docs/pages/setup/reference/authentication.mdx:17]: I'm not sure if this is still accurate. 

@codingllama is OTP still the default MFA option?

**Review by zmb3 — CHANGES_REQUESTED** [review]

File: docs/pages/setup/reference/authentication.mdx (hunk 3 above)
Comment (codingllama) [reply docs/pages/setup/reference/authentication.mdx:17]: Correct, "otp" is the default for local second_factor settings. It makes sense, as there is nothing to configure for OTP ("webauthn" needs an RPID, at least).

File: docs/pages/setup/reference/authentication.mdx (hunk 2 above)
Comment (ptgott) [reply docs/pages/setup/reference/authentication.mdx:10]: It looks like a couple other parts of the Teleport docs/product mention a "local" auth connector. 

- [Enterprise SSO guide](https://github.com/gravitational/teleport/blame/75e24cbd54ccb7850d257be1dad59208a4179d38/docs/pages/enterprise/sso.mdx#L61)
- The help text for the `auth` flag of `tsh` (https://goteleport.com/docs/setup/reference/cli/#tsh) seems to imply that `local` is a kind of auth connector

Should we be more careful here and in those other locations to distinguish `local` form actual auth connectors? 

For this PR, I've suggested changing this H2 to "Local (no authentication connector)".

Comment (zmb3) [reply docs/pages/setup/reference/authentication.mdx]: Yes, I think we should be more careful throughout the docs.

If you go to the "auth connectors" tab of the web UI, you won't see one for local, so referring to it as the "Local Auth Connector" is likely to confuse people.

**Review by zmb3 — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

File: docs/pages/setup/reference/authentication.mdx (hunk 2 above)
Comment (ptgott) [reply docs/pages/setup/reference/authentication.mdx]: 👍 Added an issue: https://github.com/gravitational/teleport/issues/10761

**Pull request merged**